package.json

```diff
@@ -91,6 +91,10 @@
       "es6": true,
       "node": true
     },
+    "extends": [
+      "eslint:recommended",
+      "tjw-jsdoc"
```

Comment:
I realise I didn't enable the jsdoc eslint plugin here. Not too familiar with `tjw-jsdoc`. Any notable differences between `tjw-jsdoc` and `jsdoc` plugin?

Reply:
PR 1 and PR 2 are just about *how* to set up the rules.

PR 3 is about applying those rules to the codebase and updating files. If there is something you wish was handled different with how the JSDoc blocks are formatted, or how strictly enforced they are, you can comment in PR 3 what changes you'd want and I'll try to find the rule/setting to adjust for that. As it is right now, there's a lot of comment blocks that need descriptions added to explain what a function is doing (like 200ish). So it's a lot of work to document every single thing the library does, and can be very tedious. It does make the code base much easier to work in for new people though, you don't have to read a 20 or 80 line function to understand what it does, the comment above it will always tell you what it does in a quick sentence, what it takes in, and what it returns. But again, that's a nice thing for new people, but a ton of work to set up all at once. Going through and just documenting the types of everything will take a while, but will at least give development benefits as a TS engine will highlight things it doesn't like then.

Or you can skip all this, and just use JSDocs to define out the API layer to give types to other repos. This is equivelant to having a `.d.ts` file in the root of the repo, except [JSDocs gives much better results than TypeScript for API interfaces](https://github.com/nwutils/create-desktop-shortcuts/pull/69).

Reply:
> This is why I did two PR's comparing. If you 100% agree with me on everything (or don't care), then this is the simpler approach, you offload caring about the settings to someone else.

I'm honestly fine with this.

> However, if there are any parts of it you disagree with, you can simply add that rule into the rules section here and override those specific things, while still inherriting the rest of my rules.

As I learn more about JSDoc, tweaks can be made when required. This looks like a good start!

> The point is that eslint-plugin-jsdoc has a ton of options, but by default almost all of them aren't turned on. This turns on most of them and adjusts their settings to my liking. I just don't want to force my opinions on your repo.

I know that you know more about JSDoc and JavaScript than me and for me it's not about forcing your opinion on the repo but me learning some new. My first coding gig involved TypeScript and I learnt (and am still learning) the quirks of JavaScript after it. I've come to realise that where TypeScript is a better fit and where JavaScript be be better utilised!

> As it is right now, there's a lot of comment blocks that need descriptions added to explain what a function is doing (like 200ish). So it's a lot of work to document every single thing the library does, and can be very tedious.

The approach I was thinking of was adding `eslint-config-tjw-jsdoc` to the eslint config via this PR, turn off linting in ci and gradually add the jsdocs whenever a code change made lacks it. Let me know what you think?

>  It does make the code base much easier to work in for new people though, you don't have to read a 20 or 80 line function to understand what it does, the comment above it will always tell you what it does in a quick sentence, what it takes in, and what it returns.

Completely agree. The number and quality old contributions would definitely go up because of it. It would also be easier to maintain for the next maintainer(s) after me.